meta/dataset.go

```diff
@@ -0,0 +1,112 @@
+package meta
+
+import (
+	"sort"
+	"strconv"
+	"strings"
+)
+
+const (
+	datasetDomain int = iota
+	datasetNetwork
+	datasetKey
+	datasetTilde
+	datasetLast
+)
+
+var datasetHeaders Header = map[string]int{
+	"Domain":  datasetDomain,
+	"Network": datasetNetwork,
+	"Key":     datasetKey,
+	"Tilde":   datasetTilde,
+}
+
+type Dataset struct {
+	Domain  string
+	Network string
+	Key     string
+	Tilde   bool
+
+	tilde string
```

Comment:
question: assume this is because when decoding, you want to produce the exact same input, i.e. "True" -> "True", "TRUE" -> "TRUE" etc.?

Reply:
It's a shame Go doesn't allow for optional parentheses for no argument functions, otherwise you could make `Tilde` be a function that returns bool based on the value of `tilde`. But since it is required, making it into a function is a little ugly, as you'd be calling `d.Tilde()`, and other values `d.Domain`. You do get to remove a mutable variable though.
Still, I'm happy with this, thanks for the context.